### ISSUE TITLE
sharing GPX files

### ISSUE THREAD

**Issue by RicoZ-OSM** [jonasoreland/runnerup#426]
It seems that RunnerUp does share an "application/vnd.garmin.workout+json" which has the gpx as an extra.

Other programs may instead support receiving intents like
`
 ACTION:android.intent.action.VIEW
 DATA:content://com.fsck.k9.attachmentprovider/7701a9c7-0d95-4477-b64a-f84a96d19f24/81/VIEW
 TYPE:application/gpx+xml
`
Could that be added as an extra option?

See also https://github.com/osmandapp/Osmand/issues/2896

Btw sharing tracks to OsmAnd makes for a great map visualizer using the offline vector maps and even satelite tile overlay if wanted.

**Comment by gerhardol** [comment]
RunnerUp shares ShareActivity intent with ShareWorkout. Could probably be changed. I may take a look what changing gives.

While the current mime seem to be incorrect, there may be better options than the suggested.
The Garmin intent type is "application/vnd.garmin.tcx+xml" (Garmin has defined similar for gpx), should probably be used instead. If there is a benefit in using "application/gpx+xml" for GPX that should be used for gpx. A quick search do not give any conclusion.

**Comment by gerhardol** [comment]
I tried with both application/vnd.garmin.*+xml and application/*+xml
However I have no application specifically handling these mime types.
The only application I have found (not searching too hard) handling application/gpx+xml is GPX Viewer in Play. However, that app do not support content:/STREAM as RunnerUp sends them.

The "content:" problem is something other apps have too. For instance Dropbox, Gmail, Solid Explorer works fine but ES Explorer do not. Permission changes in Nougat may force other apps to change too from what I understand.

I will make a pull request to change the mime type to application/gpx+xml and application/vnd.garmin.tcx+xml.
